sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreChangelog.scala

```diff
@@ -155,6 +161,11 @@ class StateStoreChangelogWriterV1(
       operationName = "Delete", entity = "changelog writer v1")
   }
 
+  override def merge(key: Array[Byte], value: Array[Byte], colFamilyName: String): Unit = {
+    throw new UnsupportedOperationException("Operation not supported with state " +
```

Comment:
Do we need to integrate with NERF here ?

Reply:
ChangelogCheckpointingWriterV1 currently throws UnsupportedOperationException for all column family based operations. I followed the same approach here. Would the user ever hit this directly? We have guardrails in state store initialization to ensure such a scenario never happens, right?